pkg/prop/prop.go

```diff
@@ -119,11 +119,9 @@ type Codec struct {
 	// Target bitrate in bps.
 	BitRate int
 
-	// Quolity of the encoding [0-9].
-	// Larger value results higher quality and higher CPU usage.
-	// It depends on the selected codec.
-	Quality int
-
 	// Expected interval of the keyframes in frames.
 	KeyFrameInterval int
+
+	// Library specific parameter struct defined by each codec package
+	CodecParams interface{}
```

Comment:
maybe better to make it 
```go
interface{
  Default() error
}
```
to force having default value loader.

Reply:
But, it can't handle parameter mutation of vpx's VP8/9.
libvpx uses one config struct both for VP8 and VP9. Current implementation is better for this case.
```go
cp, err := vpx.NewVP8Param()
if err != nil {
  panic(err)
}
p.CodecParams = cp
```

```go
var cp vpx.Param // type must be added per VP8/VP9 to have LoadDefault interface.
if err := cp.LoadDefault(); err != nil {
  panic(err)
}
p.CodecParams = cp
```

Let me keep the current interface.

Reply:
I think I like the way it is. If `CodecParam`, the caller and the codec can assume that it'll use default values.